lib/components/Portal/PortalProvider.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {
+	Context,
+	createContext,
+	FunctionComponent,
+	RefObject,
+	useContext,
+	useRef,
+} from 'react';
+
+interface Props {
+	children;
+}
+
+export const PortalContext: Context<{
+	portalInstanceRef: RefObject<HTMLDivElement>;
+}> = createContext(null);
+
+export const usePortalContext = () => useContext(PortalContext);
+
+const buildPortalElement = (): HTMLDivElement => {
+	if (typeof window === 'undefined') return null;
+
+	const element: HTMLDivElement = document.createElement('div');
+	element.setAttribute('id', `od-portal-${Math.round(Math.random() * 1e6)}`);
+	document.body.append(element);
+
+	return element;
+};
+
+export const PortalProvider: FunctionComponent<Props> = ({ children }) => {
```

Comment:
I think you might have misunderstood me. The PortalProvider isnt a "component" in the traditional sense.

If you look at our [BulletList](https://github.com/autoguru-au/overdrive/blob/cd60629370e235849c5bbf13afc05ff955c402a0/lib/components/BulletList/BulletList.tsx#L16-L29) component, you'll see how the provider is created as a child to the Component.

So in a sense you'll go:

```jsx
export const Portal = ({ children }) => {
	const maybeInstance = useContext(portalContext);
	const rootDivRef = useRef();

	return createPortal(<div ref={rootDivRef}>
		<portalContext.Provider value={rootDivRef}>
			{children}
		</portalContext.Provider>
	</div>, maybeInstance ?? document.body);
};
```

This way there is no root provider needed to be imported by everyone, no invariant check, and no manual div creation.